tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -0,0 +1,657 @@
+# Copyright 2018 Google LLC. All rights reserved. Licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+# Any software provided by Google hereunder is distributed "AS IS", WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, and is not intended for production use.
+"""Script to move a bucket, all settings and data from one project to another."""
+
+from __future__ import absolute_import
+import datetime
+from time import sleep
+
+from retrying import retry
+from yaspin import yaspin
+
+from google.cloud import exceptions
+from google.cloud import pubsub
+from google.cloud import storage
+from google.cloud.storage import iam
+from googleapiclient import discovery
+
+from gcs_bucket_mover import bucket_details
+from gcs_bucket_mover import configuration
+from gcs_bucket_mover import sts_job_status
+
+CHECKMARK = u'\u2713'.encode('utf8')
+
+
+def move_bucket(conf):
+    """Main entry point for the bucket mover script
+
+    Args:
+        conf: the argparser parsing of command line options
+    """
+
+    #Load the environment config values set in config.sh and create the storage clients.
+    config = configuration.Configuration(conf)
+
+    # Get copies of all of the source bucket's IAM, ACLs and settings so they can be copied over to
+    # the target project bucket
+    # Can't list Object Change Notifications
+    source_bucket = config.source_storage_client.lookup_bucket(
+        config.bucket_name)
+    source_bucket_details = bucket_details.BucketDetails(
+        conf=conf, source_bucket=source_bucket)
+
+    # The notifications list method returns an iterator, so we must go through it to get a copy of
+    # all of the items
+    # Don't append directly to the source_bucket_details because the setter is used to check if
```

Comment:
I don't understand this comment. I think it shouldn't be here because it is a comment on the internals of source_bucket_details

Reply:
Moved this into bucket_details.py.